tests/s3/versioning/test_get_put_object_tagging.py

```diff
@@ -695,3 +695,237 @@ def test_get_put_obj_tags_susp_bkt_40432(self):
                                                     f"Expected: {put_tag} \n Actual: {get_tag}")
 
         LOGGER.info("ENDED: Test GET and PUT object tagging in a versioning suspended bucket")
+
+    @pytest.mark.s3_ops
+    @pytest.mark.tags("TEST-40433")
+    def test_get_put_obj_tags_del_bkt_40433(self):
+        """Test GET and PUT object tagging for deleted versioned object"""
+
+        LOGGER.info("STARTED: Test GET and PUT object tagging for deleted versioned object")
+
+        LOGGER.info("Step 1: Perform PUT Bucket versioning with status as Enabled on %s",
+                    self.bucket_name)
+        resp = self.s3_ver_obj.put_bucket_versioning(bucket_name=self.bucket_name)
+        assert_utils.assert_true(resp[0], resp)
+
+        LOGGER.info("Step 2: Upload Object %s with version enabled bucket %s",
+                    self.object_name, self.bucket_name)
+        s3_cmn_lib.upload_version(self.s3_test_obj, bucket_name=self.bucket_name,
+                                  file_path=self.file_path, object_name=self.object_name,
+                                  versions_dict=self.versions)
+        latest_ver_id = self.versions[self.object_name]["version_history"][-1]
+
+        LOGGER.info("Step 3: Perform PUT Object Tagging for %s with a tag key-value "
+                    "pair", self.object_name)
+        resp = s3_cmn_lib.put_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
+                                             s3_ver_test_obj=self.s3_ver_obj,
+                                             bucket_name=self.bucket_name,
+                                             object_name=self.object_name,
+                                             version_tag=self.ver_tag, versions_dict=self.versions)
```

Comment:
don't we want to mention version_id?

Reply:
according to step mentioned in TD, we don't have to mention version_id.
Step:
Perform PUT Object Tagging for object1 with a tag key-value pair